### DISCUSSION TITLE
Check if HL7v2 Segment Exists

### DISCUSSION THREAD

**Discussion by JJWest12** [nextgenhealthcare/connect#5913] [Q&A] [answered]
I have some code that is checked OBX.15 for a value. I am running in to some issue with messages that do not have an OBX.15 segment in them. What is the most efficient way of checking to see if the segment even exists?

    var id = ''
    if(msg['OBX'][0]['OBX.15'].toString() != ''){
        id = msg['OBX'][0]['OBX.15']['OBX.15.1'].toString()
    }

**Comment by pacmano1** [comment]
@JJWest12 that question is been asked many times over in the forums, there is still some great stuff over there.  See https://forums.mirthproject.io/forum/mirth-connect/support/8416-check-if-hl7-segments-exist among a few other posts.

**Reply by pacmano1** [reply, marked as answer]
Having said that, you might find 
https://github.com/MichaelLeeHobbs/mmc/blob/7b7fbfc7b1845b16bc7d364802d146d0c857f253/src/codeTempaltes/Globals/%24t.js#L6 invaluable.  Add that as a code template for all channels.

Effectively it lets you do stuff like:

`$c('test', $t(() => msg['OBX'][0]['OBX.15']['OBX.15.1'].toString()) || '')`

If that segment does not exist, the channelMap var gets set to `''` (an empty string), otherwise it gets set to the current value.
